integration-libs/cdc/core/auth/services/user-authentication/cdc-login.service.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { TestBed, waitForAsync } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { RouterTestingModule } from '@angular/router/testing';
+import {
+    AuthService,
+    GlobalMessageService,
+    I18nTestingModule,
+    WindowRef
+} from '@spartacus/core';
+import { LoginFormComponentService } from 'feature-libs/user/account/components/login-form/login-form-component.service';
+import { CdcJsService } from 'integration-libs/cdc/root';
+import { FormErrorsModule } from 'projects/storefrontlib/shared';
+import { of } from 'rxjs';
+import { CdcLoginComponentService } from './cdc-login.service';
+import createSpy = jasmine.createSpy;
+
+class MockWinRef {
+  get nativeWindow(): Window {
+    return {} as Window;
+  }
+}
+
+class MockAuthService implements Partial<AuthService> {
+  loginWithCredentials = createSpy().and.returnValue(of({}));
+  isUserLoggedIn = createSpy().and.returnValue(of(true));
+}
+
+class MockGlobalMessageService {
+  add = createSpy().and.stub();
+  remove = createSpy().and.stub();
+}
+
+class MockCDCJsServiceWithCDC implements Partial<CdcJsService> {
+  didLoad = createSpy().and.returnValue(of(true));
+  registerUserWithoutScreenSet = createSpy().and.callFake((user) => of(user));
```

Comment:
please type `user`

Reply:
did you mean type of user like `user: UserSignup`?

thanks

Reply:
yes